Tests/MongoSwiftTests/AsyncAwaitTestUtils.swift

```diff
@@ -0,0 +1,141 @@
+    sleepInterval: TimeInterval = 0.1,
+    _ block: () -> Bool
+) async throws {
+    let start = DispatchTime.now()
+    while DispatchTime.now() < start + timeout {
+        if block() {
+            return
+        }
+        try await Task.sleep(seconds: sleepInterval)
+    }
+    XCTFail("Expected condition \"\(description)\" to be true within \(timeout) seconds, but was not")
+}
+
+extension MongoSwiftTestCase {
+    internal func withTestClient<T>(
+        _ uri: String = MongoSwiftTestCase.getConnectionString().toString(),
+        options: MongoClientOptions? = nil,
+        eventLoopGroup: EventLoopGroup? = nil,
+        f: (MongoClient) async throws -> T
+    ) async throws -> T {
+        let elg = eventLoopGroup ?? MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { elg.syncShutdownOrFail() }
+        let client = try MongoClient.makeTestClient(uri, eventLoopGroup: elg, options: options)
+        defer { client.syncCloseOrFail() }
+        return try await f(client)
+    }
+
+    // swiftlint:disable large_tuple
+    // see: https://github.com/realm/SwiftLint/issues/3753
+    internal func withTestNamespace<T>(
+        ns: MongoNamespace? = nil,
+        collectionOptions: CreateCollectionOptions? = nil,
+        _ f: (MongoClient, MongoDatabase, MongoCollection<BSONDocument>) async throws -> T
+    ) async throws -> T {
+        let ns = ns ?? self.getNamespace()
+        guard let collName = ns.collection else {
+            throw TestError(message: "missing collection")
+        }
+        return try await self.withTestClient { client in
+            let database = client.db(ns.db)
+            let collection: MongoCollection<BSONDocument>
+            do {
+                collection = try await database.createCollection(collName, options: collectionOptions)
+            } catch let error as MongoError.CommandError where error.code == 48 {
+                try await database.collection(collName).drop()
+                collection = try await database.createCollection(collName, options: collectionOptions)
+            }
+            defer { collection.syncDropOrFail() }
+            return try await f(client, database, collection)
+        }
+    }
+    // swiftlint:enable large_tuple
+}
+
+extension MongoClient {
+    internal func serverVersion() async throws -> ServerVersion {
+        let reply = try await self.db("admin").runCommand(
+            ["buildInfo": 1],
+            options: RunCommandOptions(readPreference: .primary)
+        )
+        guard let versionString = reply["version"]?.stringValue else {
+            throw TestError(message: " reply missing version string: \(reply)")
+        }
+        return try ServerVersion(versionString)
+    }
+
+    internal func serverParameters() async throws -> BSONDocument {
+        try await self.db("admin").runCommand(["getParameter": "*"])
+    }
+
+    internal func topologyType() async throws -> TestTopologyConfiguration {
```

Comment:
was fun to get to try out some of the new concurrency features in these next few methods and implicitly confirm the driver can handle parallel usage across tasks! (explicit test of that to come soon in another PR)